[sql] Invalid Rus with PRIMARY KEY + LATENESS

This fails to compile:

```sql
CREATE TABLE t (
    ts BIGINT NOT NULL PRIMARY KEY LATENESS 5
);
```

```
 error[E0308]: mismatched types
   --> crates/feldera_pipe_operator_23f7c113aadd244d/src/lib.rs:71:47
    |
71  |             Tup1::new(minus_i64_i64((*t_2).0, cast_to_i64_i32(5i32)))
    |                       -------------           ^^^^^^^^^^^^^^^^^^^^^ expected `i64`, found `Result<i64, Box<SqlRuntimeError>>`
    |                       |
    |                       arguments to this function are incorrect
    |
    = note: expected type `i64`
               found enum `Result<i64, Box<SqlRuntimeError>>`
```

--- a/sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/circuit/operator/DBSPInputMapWithWaterlineOperator.java
+++ b/sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/circuit/operator/DBSPInputMapWithWaterlineOperator.java
@@ -152,6 +152,18 @@ public class DBSPInputMapWithWaterlineOperator
     public void accept(InnerVisitor visitor) {
         visitor.property("originalRowType");
         this.originalRowType.accept(visitor);
+        visitor.property("outputType");
+        this.outputType.accept(visitor);
+        visitor.property("initializer");
+        this.initializer.accept(visitor);
+        visitor.property("timestamp");
+        this.timestamp.accept(visitor);
+        visitor.property("lub");
+        this.lub.accept(visitor);
+        visitor.property("filter");
+        this.filter.accept(visitor);
+        visitor.property("error");
+        this.error.accept(visitor);
     }
 
     @Override

--- a/sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/visitors/outer/CircuitRewriter.java
+++ b/sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/visitors/outer/CircuitRewriter.java
@@ -23,6 +23,7 @@
 
 package org.dbsp.sqlCompiler.compiler.visitors.outer;
 
+import org.dbsp.sqlCompiler.circuit.IMultiOutput;
 import org.dbsp.sqlCompiler.circuit.operator.DBSPAsofJoinOperator;
 import org.dbsp.sqlCompiler.circuit.operator.DBSPAggregateLinearPostprocessOperator;
 import org.dbsp.sqlCompiler.circuit.operator.DBSPAggregateLinearPostprocessRetainKeysOperator;
@@ -36,6 +37,7 @@ import org.dbsp.sqlCompiler.circuit.operator.DBSPControlledKeyFilterOperator;
 import org.dbsp.sqlCompiler.circuit.operator.DBSPFlatMapIndexOperator;
 import org.dbsp.sqlCompiler.circuit.operator.DBSPFlatMapOperator;
 import org.dbsp.sqlCompiler.circuit.operator.DBSPIndexedTopKOperator;
+import org.dbsp.sqlCompiler.circuit.operator.DBSPInputMapWithWaterlineOperator;
 import org.dbsp.sqlCompiler.circuit.operator.DBSPIntegrateTraceRetainKeysOperator;
 import org.dbsp.sqlCompiler.circuit.operator.DBSPInternOperator;
 import org.dbsp.sqlCompiler.circuit.operator.DBSPJoinFilterMapOperator;
@@ -148,6 +150,31 @@ public class CircuitRewriter extends CircuitCloneVisitor {
         } else {
             super.replace(operator);
         }
+    }
+
+    @Override
+    public void postorder(DBSPInputMapWithWaterlineOperator operator) {
+        IMultiOutput result = operator;
+        DBSPTypeStruct originalRowType = this.transform(operator.originalRowType).to(DBSPTypeStruct.class);
+        DBSPTypeIndexedZSet outputType = this.transform(operator.outputType(0)).to(DBSPTypeIndexedZSet.class);
+        DBSPClosureExpression initializer = this.transform(operator.initializer).to(DBSPClosureExpression.class);
+        DBSPClosureExpression timestamp = this.transform(operator.timestamp).to(DBSPClosureExpression.class);
+        DBSPClosureExpression lub = this.transform(operator.lub).to(DBSPClosureExpression.class);
+        DBSPClosureExpression filter = this.transform(operator.filter).to(DBSPClosureExpression.class);
+        DBSPClosureExpression error = this.transform(operator.error).to(DBSPClosureExpression.class);
+        if (!originalRowType.sameType(operator.originalRowType) ||
+            !outputType.sameType(operator.outputType(0)) ||
+            initializer != operator.initializer ||
+            timestamp != operator.timestamp ||
+            lub != operator.lub ||
+            filter != operator.filter ||
+            error != operator.error) {
+            result = new DBSPInputMapWithWaterlineOperator(
+                    operator.getRelNode(), operator.sourceName, operator.keyFields,
+                    outputType, originalRowType, operator.metadata,
+                    operator.tableName, initializer, timestamp, lub, filter, error);
+        }
+        this.map(operator, result, true);
     }
 
     @Override

--- a/sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/Regression1Tests.java
+++ b/sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/Regression1Tests.java
@@ -912,4 +912,9 @@ public class Regression1Tests extends SqlIoTest {
                  true | 1
                  false | 1""");
     }
+
+    @Test
+    public void issue4729() {
+        this.getCCS("CREATE TABLE t (ts BIGINT NOT NULL PRIMARY KEY LATENESS 5);");
+    }
 }

